File: Easy/226. Invert Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {

  let traverse = (branch) => {
      if (!branch) {
          return;
      }

      let temp = branch.left;
      branch.left = branch.right;
      branch.right = temp;
      traverse(branch.left);
      traverse(branch.right);
  }
  traverse(root)
  return root;
};

File: Easy/226. Invert Binary Tree.py
class Solution(object):
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        # if root:
        # invert = self.invertTree
        # root.left, root.right = invert(root.right), invert(root.left)
        # return root
        if root:
            left = root.left
            root.left = root.right
            root.right = left
            self.invertTree(root.left)
            self.invertTree(root.right)
            return root